src/hotspot/os/posix/os_posix.cpp

```diff
@@ -1339,6 +1339,13 @@ static jlong millis_to_nanos_bounded(jlong millis) {
   return millis_to_nanos(millis);
 }
 
+static jlong nanos_to_nanos_bounded(jlong nanos) {
```

Comment:
I don't think we actually need this for nanos. There is no overflow potential and the MAX_SECS limit will be handled later.

Reply:
Hm, "later" where? Because I see the call to `pthread_condwait_timed` right after we called `nanos_to_nanos_bounded`, and this looks like a last opportunity to handle the `MAX_SECS` limit:
https://github.com/openjdk/jdk/pull/13225/files#diff-7daa601d72ef74e5281faf8256b537b4ec9c5e5c236b716902592da12fa2aad2R1582

Reply:
`to_abst_time` will call ` calc_rel_time` which checks for MAX_SECS.